### PR TITLE
Add support for predicting class probabilities

### PR DESCRIPTION
**Pull request by danielenricocahall**
Add support for class probability prediction when utilizing `ElephasTransformer`. This addresses https://github.com/maxpumperla/elephas/issues/174 and https://github.com/maxpumperla/elephas/issues/137. This also resulted in some simplification of the logic in the `_transform` method, as there were some unnecessary field type conversions.

### CODE REVIEW

File: elephas/ml_model.py [hunk 1]
```diff
@@ -178,40 +187,40 @@ def _transform(self, df):
...
+                    self.model_type,
+                    self.get_predict_classes()))
+        if (self.model_type == ModelType.CLASSIFICATION and self.get_predict_classes()) \
+                or self.model_type == ModelType.REGRESSION:
             predictions = predictions.map(lambda x: tuple([float(x)]))
+            output_col_field = StructField(output_col, DoubleType(), True)
+        else:
+            # we're doing classification and predicting class probabilities
+            predictions = predictions.map(lambda x: tuple([x.tolist()]))
+            output_col_field = StructField(output_col, ArrayType(DoubleType()), True)
```
[the remark this answers is not in the document]
Comment (danielenricocahall) [reply elephas/ml_model.py:219]: @OscarDPan  I think that's correct, although we're not doing argmax here - we only do argmax in the case we're using the keras functional API: https://github.com/maxpumperla/elephas/blob/master/elephas/utils/model_utils.py#L64. To your point, we may want to add a check there based on the loss.

[the remark this answers is not in the document]
Comment (OscarDPan) [reply elephas/ml_model.py:219]: Ah yes that's the line I meant to comment on. :)

[the remark this answers is not in the document]
Comment (OscarDPan) [reply elephas/ml_model.py:219]: My suggestion is instead of complicating Elephas's code by adding multiple checks, we can just implement the distributed predict() method and let end users decide how they want to transform the predicted probabilities.

I like Elephas for its simplicity and light weight and since there aren't many team members in this project we can focus more on the "distributed" features. Just my 2cents.

[the remark this answers is not in the document]
Comment (OscarDPan) [reply elephas/ml_model.py:219]: Hi @danielenricocahall I found an edge case.
I am using a customized loss function call "tweedie loss", it's model type is supposed to be "ModelType.REGRESSION" but since it's not in the dictionary, in the end it goes to 
```        else:
            # we're doing classification and predicting class probabilities
            predictions = predictions.map(lambda x: tuple([x.tolist()]))
            output_col_field = StructField(output_col, ArrayType(DoubleType()), True)
```

The consequence is that all my existing code assuming the output should be a float gets error, since it is a list of one float. I was wondering if we can do some dynamic schema detection s.t. if len(output) == 1 then it's a DoubleType() otherwise it's ArrayType(DoubleType())?

[the remark this answers is not in the document]
Comment (danielenricocahall) [reply elephas/ml_model.py:219]: Ah! You can register the loss using:

`LossModelTypeMapper().register_loss('tweedie_loss', ModelType.REGRESSION)`

[the remark this answers is not in the document]
Comment (danielenricocahall) [reply elephas/ml_model.py:219]: Regarding your other suggestion of just making a distributed `predict`, I like it - however, that change may warrant a new major release, since any applications depending on `predict` and expect it to output a singular value would have to update their code (it's a breaking change).

[the remark this answers is not in the document]
Comment (OscarDPan) [reply elephas/ml_model.py:219]: Cool! And thanks for the API tip!

**Review by OscarDPan — APPROVED** [review]
